### PR TITLE
#162948851 Pagination

### PR DESCRIPTION
**Pull request by MaggieChege**
**What does this PR do?**
Adds Pagination to articles

**Description of Task to be completed?**

Articles and Comments are paginated 

**How should this be manually tested?**

Clone the repo locally

cd ah-django-unchained
git checkout ft-pagination-162948851
get started with the app as stated in the README section

**What are the relevant pivotal tracker stories?**

[[#162948851]](https://www.pivotaltracker.com/story/show/162948851)

Screenshots
image

### CODE REVIEW

**Review by KMaina — APPROVED** [review]
Looks good to me

**Review by d-kahara — APPROVED** [review]
Looks good to me

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -15,17 +13,19 @@
...
 from .serializers import (ArticleSerializer,
                           GetArticleSerializer, DeleteArticleSerializer,
                           RatingSerializer, CommentSerializer,
                           DeleteCommentSerializer)
 from .models import (Article, ArticleRating, Comment)
+from .pagination import CustomPagination
 
 
 class ArticleAPIView(ListCreateAPIView):
+    pagination_class = CustomPagination
```
Comment (ignatiusukwuoma) [inline authors/apps/articles/views.py]: This should come after docstrings

File: authors/apps/articles/pagination.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from rest_framework.pagination import PageNumberPagination
+
+
+class CustomPagination(PageNumberPagination):
+    page_size = 10
+    page_size_query_param = 'page_size'
+    max_page_size = 100
```
Comment (ignatiusukwuoma) [inline authors/apps/articles/pagination.py:10]: Can you explain `page size` and `max_page_size`? Add brief explanation as a comment on the class. I also see a PAGE_SIZE=100 in settings, where does that fall into?

Comment (MaggieChege) [reply authors/apps/articles/pagination.py:10]: Okay

File: authors/apps/articles/views.py [hunk 3]
```diff
@@ -15,18 +13,21 @@
...
                           RatingSerializer, CommentSerializer,
                           DeleteCommentSerializer)
 from .models import (Article, ArticleRating, Comment)
+from .pagination import CustomPagination
 
 
 class ArticleAPIView(ListCreateAPIView):
     """Creates articles and retrieves all articles"""
+
+    pagination_class = CustomPagination
```
Comment (ignatiusukwuoma) [inline authors/apps/articles/views.py:30]: Articles are not ordered in any way so pagination may yield inconsistent results. How about you add an ordering for the list of articles?

Comment (MaggieChege) [reply authors/apps/articles/views.py:30]: Okay, let me add that.

**Pull request merged**